helper/result_string_search.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Search for the benchmarks that contains a specific string in result.
+
+Example usage:
+To search for benchmarks that failed to parse error:
+python3 -m helper.result_string_search -r results -s "<error>
+
+</error>"
+
+Optionally:
+[--sub] provide subdirectory name to limit the scope of search.
+[--url] provide template to output http links for easy access to web report.
+"""
+import argparse
+import logging
+import os
+
+
+def parse_args() -> argparse.Namespace:
+  """parse arguments"""
+  parser = argparse.ArgumentParser(
+      description=
+      'Search for all benchmark that contains the <string> in <result>')
+  parser.add_argument('-s',
+                      '--string',
+                      type=str,
+                      required=True,
+                      help='The string to search in result.')
+  parser.add_argument('-r',
+                      '--result',
+                      type=str,
+                      required=True,
+                      help='The root path to the result directory.')
+  parser.add_argument(
+      '-b',
+      '--sub',
+      type=str,
+      default='',
+      help=('The subdirectory to search in each output-* directories. '
+            'Search in all subdirectories by default'))
+  parser.add_argument(
+      '-u',
+      '--url',
+      type=str,
+      default='',
+      help='Optional url template to web report for easy access.')
+
+  args = parser.parse_args()
+  assert os.path.isdir(args.result), '--result must be an existing directory.'
+
+  output_dir = next(os.walk(args.result))[1][0]
```

Comment:
If `output_dir` is expected to be a direct child dir (e.g., benchmark's output directory), could we use `os.listdir(args.result)`?

Correct me if I am wrong: `next` merely gets the first element from the iterator, which is the same as `[0]`. In that case, could we use `[0]` for clarity?

Taking one step further, I guess the `assert` below:
1. Does not directly check if `args.sub` is one of `[fixed_targets, logs, raw_targets, ..., status]` for flexibility, e.g., some benchmarks do not have them due to exp failures, or we added more sub dirs later.
2. Checks if `args.sub` in the *first* (not other) benchmark result dir for simplicity, assuming all benchmark result dirs have the same sub dirs.
In this case, would it be better to assert *at least one of the benchmark result dirs contain `sub`* for better flexibility without losing simplicity?  E.g., use `any`.

Reply:
Using `next` on `os.walk()` because it returns a generator instead of an iterator, so we can't get the element at a certain index. We can even modify the result on the fly, e.g. [ignore corpora](https://github.com/google/oss-fuzz-gen/pull/136/files#diff-74f98584d5ba085369bd1b5bca0319fa225807042fc53f1acd67732aa7acd2ccR117)

Both assumptions are true. Will use any, thanks :)